### PR TITLE
Fix unit translation for recurring contributions

### PR DESCRIPTION
**Pull request by masetto**
I worked to extend @mlutfy pull request [#19655](https://github.com/civicrm/civicrm-core/pull/19655) on confirmation page and for recurring contributions. 
Also I fixed some code indentations.

### CODE REVIEW

File: templates/CRM/Contribute/Form/Contribution/Confirm.tpl [hunk 1]
```diff
@@ -111,10 +111,30 @@
...
-                  <strong>{ts 1=$frequency_interval 2=$frequency_unit}I want to contribute this amount every %1 %2s.{/ts}</strong>
+                  {if $frequency_unit eq 'day'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 days.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'week'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 weeks.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'month'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 months.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'year'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 years.{/ts}</strong></p>
+                  {else}
```
Comment (mlutfy) [inline templates/CRM/Contribute/Form/Contribution/Confirm.tpl:122]: I think we can avoid the final "else"? #19655 does not have one.

**Comment by eileenmcnaughton** [comment]
@masetto did you spot the (minor) feedback on this?

File: templates/CRM/Contribute/Form/Contribution/Confirm.tpl (hunk 1 above)
Comment (masetto) [reply templates/CRM/Contribute/Form/Contribution/Confirm.tpl:122]: Yes, I know, but I put it in as a precaution, so it works even if someone changes it or another unit is introduced in the future 😉

**Comment by demeritcowboy** [comment]
What's the status on this one? Is it just waiting for someone to have a strong opinion on whether to include an else clause or not?

My thoughts: You can add frequency units to the list in the admin UI. I'm not sure they get used properly everywhere, but then at least these pages would display it, so I'd lean towards keeping the else clause. (Note: I haven't run this, just commenting on what seems to be the outstanding question.)

**Comment by mlutfy** [comment]
Fine by me - I did not know that we could add units! (I suspect that it might cause unknown behaviours?)

I think it would need an `r-run`, just as a precaution, and from reading the code, I think it would generate a `<p>`  twice?

File: templates/CRM/Contribute/Form/Contribution/Confirm.tpl [hunk 2]
```diff
@@ -111,10 +111,30 @@
             {else}
               {if $frequency_interval > 1}
                 <p>
```
Comment (mlutfy) [inline templates/CRM/Contribute/Form/Contribution/Confirm.tpl:113]: i.e. do we need to remove this p tag (and the associated closing tag)?

Comment (demeritcowboy) [reply templates/CRM/Contribute/Form/Contribution/Confirm.tpl:113]: I did a little run and it does seem to create an extra `<p>`, which looks like it creates some visual extra blank space.

And also there's a typo - the first group all say "every every".

**Comment by masetto** [comment]
Hi, I push a [fix](https://github.com/civicrm/civicrm-core/pull/20308/commits/c38994d16a1d540c1d5560e0bae3fc392e40523c).
You are right: it is better to remove the final `else`.
I'm sorry for these rough distractions.

**Comment by demeritcowboy** [comment]
Thanks for updating. Just one last thing - could you [squash commits](https://docs.civicrm.org/dev/en/latest/tools/git/#rebasing-interactive)?

**Comment by masetto** [comment]
@demeritcowboy I tried to execute `git rebase -i upstream/master` but I cannot understand the effects (I'm a beginner with Git).
Can you help me to understand what else i have to do?
Thank you,

**Comment by demeritcowboy** [comment]
Ah ok. When you run the git rebase, it opens up an editor. You should see 3 lines - one for each commit. For the second and third, change the word `pick` to either `s` or `f`. Using `s` allows you to change the wording in the commit message in the second step after you save and exit, but if you're ok with the wording on the first line then just pick `f` and it will skip the second step.

Then at the end, you can do `git push -f`, which will upload to your branch and overwrite the 3 commits with just 1 combined commit.

**Comment by demeritcowboy** [comment]
Perfect, thanks.

**Comment by mlutfy** [comment]
Thank you @masetto @demeritcowboy !

**Comment by romymy** [comment]
Hey, I have the same question as @masetto and did some research, because unfortunately in a multilingual CiviCRM word replacement is not an option ... 

I found https://lab.civicrm.org/dev/translation/-/wikis/Pushing-new-strings-to-transifex – maybe this will help.

**Comment by mlutfy** [comment]
Hi all, strings are now updated. This means we can translate on Transfiex (https://www.transifex.com/civicrm/civicrm/) and after a day, the new civicrm.mo files will be available (https://lab.civicrm.org/dev/translation/-/wikis/Administrator-Guide#updating-translation-files).

**Pull request merged**